### PR TITLE
Add counter_game_favorites migration

### PR DESCRIPTION
**Pull request by LucasGasque**
## What this PR introduces?

Add migration to counter_game_favorites table, fix counter_game_shares syntax

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] I followed GitFlow pattern to create the branch
- [X] My code produces no warnings or errors in the console
- [X] Tests for the changes have been added / updated
- [X] Mocks fot the database have been added / updated

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Documentation update
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] CI/CD related changes
```

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/database/migrations/1639508695898-Create-CounterGameFavorites-Table.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    END IF;
+
+    RETURN NULL;
+END
+$$
+    LANGUAGE plpgsql;
+
+CREATE TRIGGER update_counter_game_favorites
+    AFTER INSERT OR DELETE
+    ON shares
```
Comment (henriqueleite42) [inline src/database/migrations/1639508695898-Create-CounterGameFavorites-Table.ts:36]: `shares` -> `favorite_games`

File: src/database/migrations/1639508695898-Create-CounterGameFavorites-Table.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    FOR EACH ROW
+EXECUTE PROCEDURE update_counter_game_favorites();`;
+
+const down = `DROP TRIGGER IF EXISTS update_counter_game_favorites ON favorites;
```
Comment (henriqueleite42) [inline src/database/migrations/1639508695898-Create-CounterGameFavorites-Table.ts:40]: `favorites` -> `favorite_games`

**Review by henriqueleite42 — CHANGES_REQUESTED** [review]

File: src/database/migrations/index.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@
```
Comment (henriqueleite42) [inline src/database/migrations/index.ts:5]: Remove changes on this file

**Review by henriqueleite42 — CHANGES_REQUESTED** [review]

File: src/database/migrations/index.ts (hunk 3 above)
Comment (LucasGasque) [reply src/database/migrations/index.ts:5]: it was a syntax fix (line 8), since  i needed to change this file anyways i added the new migration!

**Review by henriqueleite42 — APPROVED** [review]

**Pull request merged**